presto-orc/src/main/java/com/facebook/presto/orc/OrcWriter.java

```diff
@@ -515,6 +517,14 @@ public void validate(OrcDataSource input)
                 orcEncoding);
     }
 
+    private int estimateAverageLogicalSizePerRow(Page page)
+    {
+        checkArgument(page.getPositionCount() > 0);
+        // sample at most 100 rows to estimate average row logical size
+        Page chunk = page.getRegion(0, Math.min(page.getPositionCount(), 100));
```

Comment:
How confident we are that the sampling is needed? Have we tried to benchmark it without sampling (taking the whole page for the average row size estimation?)

Reply:
@arhimondr : @picrinite 's following comment seems to suggest sampling is mandatory to reduce CPU:
```
Also tried not to sample but calculate total size for input page instead.
However, no improvement observed suggesting the original binary
search loop is not the root cause.
```

Reply:
@arhimondr  I tried calculation using whole page first but it didn't help.